homeassistant/components/light/lw12wifi.py

```diff
@@ -0,0 +1,215 @@
+                 transition):
+        """Initialisation of LW-12 WiFi LED Controller.
+
+        Args:
+            name: Friendly name for this platform to use.
+            host: Hostname or IP address of the device.
+            port: Port (Default: 5000) to connect to.
+            rgb_color: Initial color to set after the light turned on.
+            brightness: Brightness of the LEDs to set.
+            effect: If not None, turn on the lights with the selected effect.
+            transition: Speed of the effects.
+        """
+        import lw12
+
+        self._light = lw12.LW12Controller(host, port)
+        self._name = name
+        self._host = host
+        self._port = port
+        self._rgb_color = rgb_color
+        self._brightness = brightness
+        self._state = None
+        self._effect = effect
+        self._transition_speed = transition
+        # Setup feature list
+        self._supported_features = SUPPORT_BRIGHTNESS
+        self._supported_features |= SUPPORT_EFFECT
+        self._supported_features |= SUPPORT_RGB_COLOR
+        self._supported_features |= SUPPORT_TRANSITION
+
+    @property
+    def name(self):
+        """Return the display name of the controlled light."""
+        return self._name
+
+    @property
+    def brightness(self):
+        """Return the brightness of the light."""
+        return self._brightness
+
+    @property
+    def rgb_color(self):
+        """Read back the color of the light.
+
+        Returns [r, g, b] list with values in range of 0-255.
+        """
+        return self._rgb_color
+
+    @property
+    def effect(self):
+        """Return current light effect."""
+        if self._effect is None:
+            return None
+        return self._effect.replace('_', ' ').title()
+
+    @property
+    def is_on(self):
+        """Return true if light is on."""
+        return self._state
+
+    @property
+    def supported_features(self):
+        """Return a list of supported features."""
+        return self._supported_features
+
+    @property
+    def effect_list(self):
+        """Return a list of available effects.
+
+        Use the Enum element name for display.
+        """
+        import lw12
+        return [effect.name.replace('_', ' ').title()
+                for effect in lw12.LW12_EFFECT]
+
+    @property
+    def assumed_state(self) -> bool:
+        """Return True if unable to access real state of the entity."""
+        return True
+
+    def turn_on(self, **kwargs):
+        """Instruct the light to turn on."""
+        import lw12
+        self._light.light_on()
+        if self._effect is not None:
+            if self._effect.replace(' ', '_').upper() in lw12.LW12_EFFECT:
+                kwargs['effect'] = self._effect
+            else:
+                # Unknown effect, changing to no selected effect.
+                self._effect = None
+
+        if ATTR_RGB_COLOR in kwargs:
```

Comment:
As a heads-up, we've just merged a big refactoring to how light platforms implement colors. You'll need to rebase on the latest dev. Platforms are now going to receive turn_on reqeusts with a hue/sat color, and need to report their current color in hue/sat.

Some examples of how to convert to/from RGB if that's the preferred platform interface are here:

https://github.com/home-assistant/home-assistant/blob/947218d51c94566feeb32c6f549493fa07683252/homeassistant/components/light/hyperion.py#L142-L143

https://github.com/home-assistant/home-assistant/blob/947218d51c94566feeb32c6f549493fa07683252/homeassistant/components/light/hyperion.py#L110-L113

Reply:
Thanks you this information. I will update the code to match the new requirements.